### PR TITLE
[X] Detect empty property value in markups

### LINKED ISSUE #7187: [Bug] Typo in markup extension should throw XPE
### Description

I made a typo while writing a test case
```xaml
<Label x:Name="l1" Text="{OnPlatform iOS=foo, Default==bar}" />
```

notice the `==` instead of `=`

and got a ArgumentNullException, instead fo a proper XPE

### Steps to Reproduce

1. 
2. 
3. 

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

[X] Detect empty property value in markups

### Issues Resolved ### 

- fixes #7187

### API Changes ###

 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by StephaneDelcroix** [comment]
@samhouts should be fixed

**Comment by jfversluis** [comment]
Still fails with the same error

File: Xamarin.Forms.Build.Tasks/NodeILExtensions.cs [hunk 1]
```diff
@@ -136,12 +136,11 @@ public static bool CanConvertValue(this ValueNode node, ILContext context, TypeR
 			bool boxValueTypes, bool unboxValueTypes)
 		{
 			var module = context.Body.Method.Module;
-			var str = (string)node.Value;
-
+			if (!(node.Value is string str))
+				throw new Exception();
```
Comment (jfversluis) [inline Xamarin.Forms.Build.Tasks/NodeILExtensions.cs]: Should this have a description?

**Comment by StephaneDelcroix** [comment]
Shouldn’t be there. I added that for debug purposes.

**Comment by StephaneDelcroix** [comment]
@jfversluis fixed. thanks for noticing

**Comment by StephaneDelcroix** [comment]
@samhouts fixed. please re-review

**Comment by StephaneDelcroix** [comment]
build errors unrelated

**Pull request merged**